CSharpSourceCode/Battle/AI/CustomAIMissionLogic.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System.Linq;
+using TaleWorlds.MountAndBlade;
+using TOW_Core.Battle.AI.Components;
+using TOW_Core.Battle.Extensions;
+using TOW_Core.Utilities;
+
+namespace TOW_Core.Battle.AI
+{
+    public class CustomAiMissionLogic : MissionLogic
+    {
+        public override void OnAgentCreated(Agent agent)
+        {
+            base.OnAgentCreated(agent);
+            if (agent != Agent.Main && agent.IsAbilityUser())
+            {
+                var toRemove = agent.Components.OfType<HumanAIComponent>().ToList();
```

Comment:
if for some reason you still need to do this (see previous comment), you can use an existing utilityfunction for this from Utilities.ListExtensions RemoveAllOfType<> i think it does the same thing you want.

Reply:
I will use the utility method. Thanks!

Reply:
I had to keep the old way of doing it. It is a read only list and the values can only be modified using the Agent's RemoveComponent method.